File: android/app/src/main/java/com/autologin/wifi/WifiAccount.java
package com.autologin.wifi;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;

class WifiAccount {
  public WifiAccount(Context context) {
    preferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    username = preferences.getString(KEY_USERNAME, null);
    password = preferences.getString(KEY_PASSWORD, null);
    _isLogin = username != null && password != null;
  }


  public void setLoginInfo(String username, String password) {
    Editor prefEditor = preferences.edit();
    _isLogin = true;
    this.username = username;
    this.password = password;
    prefEditor.putString(KEY_USERNAME, username);
    prefEditor.putString(KEY_PASSWORD, password);
    prefEditor.apply();
  }

  public void clearLogin() {
    Editor prefEditor = preferences.edit();
    _isLogin = false;
    prefEditor.putString(KEY_USERNAME, null);
    prefEditor.putString(KEY_PASSWORD, null);
    prefEditor.apply();
  }

  public String getUsername() {
    return _isLogin ? username : null;
  }

  public String getPassword() {
    return _isLogin ? password : null;
  }

  public boolean isLogin() {
    return _isLogin;
  }

  private final static String PREF_NAME = "CCULIFE_WIFI_PREF";
  private final static String KEY_USERNAME = "wifi_username";
  private final static String KEY_PASSWORD = "wifi_password";

  private SharedPreferences preferences;
  private String username;
  private String password;
  private boolean _isLogin;
}

File: android/app/src/main/kotlin/com/autologin/wifi/WifiAccount.kt
package com.autologin.wifi

import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor

class WifiAccount(context:Context) {
  private val preferences: SharedPreferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
  private var username: String? = preferences.getString(KEY_USERNAME, null)
  private var password: String? = preferences.getString(KEY_PASSWORD, null)
  public var isLogin: Boolean = username != null && password != null
    private set

  fun setLoginInfo(username: String, password: String) {
    val prefEditor = preferences.edit()
    isLogin = true
    this.username = username
    this.password = password
    prefEditor.putString(KEY_USERNAME, username)
    prefEditor.putString(KEY_PASSWORD, password)
    prefEditor.apply()
  }

  fun clearLogin() {
    val prefEditor = preferences.edit()
    this.isLogin = false
    prefEditor.putString(KEY_USERNAME, null)
    prefEditor.putString(KEY_PASSWORD, null)
    prefEditor.apply()
  }

  fun getUsername() = if (isLogin) username else null

  fun getPassword() = if (isLogin) password else null

  companion object {
    val PREF_NAME = "CCULIFE_WIFI_PREF"
    val KEY_USERNAME = "wifi_username"
    val KEY_PASSWORD = "wifi_password"
  }
}
